app/cdap/components/StudioV2/components/DAGEditor/index.tsx

```diff
@@ -0,0 +1,481 @@
+      handleType = 'condition-false';
+    } else if (isSplitterEdge) {
+      sourceHandle = `source-port-${sourceNode.id}-${conn.port}`;
+      handleType = 'splitter-port';
+    }
+
+    if (sourceNode.type === 'condition') {
+      if (handleType === 'default') {
+        edgeType = 'condition-true';
+      } else if (handleType === 'condition-false') {
+        edgeType = 'condition-false';
+      }
+    }
+
+    return {
+      ...edge,
+      type: edgeType,
+      sourceHandle,
+      data: {
+        isSourceAtBottom: isAlertEdge || isErrorEdge || isConditionEdge,
+      },
+    };
+  }
+
+  function fitToScreen() {
+    reactflow.fitView({
+      padding: 20,
+      minZoom: MIN_ZOOM,
+      maxZoom: MAX_ZOOM,
+    });
+  }
+
+  useEffect(() => {
+    document.body.classList.add('with-new-dag-editor');
+    reactflow.zoomTo(0.5);
+    fitToScreen();
+
+    return () => {
+      document.body.classList.remove('with-new-dag-editor');
+    };
+  }, []);
+
+  useEffect(() => {
+    if (previewMode) {
+      document.body.classList.add('preview-mode');
+    }
+
+    return () => {
+      document.body.classList.remove('preview-mode');
+    };
+  }, [previewMode]);
+
+  useEffect(fitToScreen, [uiAutoLayout]);
+
+  function handleReconnect(oldEdge, newConn) {
+    edgeReconnectSuccessful.current = true;
+    onReconnect(oldEdge, newConn);
+  }
+
+  function handleReconnectStart() {
+    edgeReconnectSuccessful.current = false;
+  }
+
+  function handleReconnectEnd(_, edge) {
+    if (!edgeReconnectSuccessful.current) {
+      onEdgesDelete([edge]);
+    }
+
+    edgeReconnectSuccessful.current = true;
+  }
+
+  return (
+    <Provider store={AvailablePluginsStore}>
+      <DAGEditorContext.Provider value={{ isDisabled }}>
+        <ReactFlow
+          id="diagram-container"
+          nodes={nodes}
+          edges={edges}
+          nodeTypes={NODE_TYPES}
+          edgeTypes={EDGE_TYPES}
+          nodesConnectable={!isDisabled}
+          onNodesChange={isDisabled ? undefined : onNodesChange}
+          onEdgesChange={isDisabled ? undefined : onEdgesChange}
+          onConnect={isDisabled ? undefined : onConnect}
+          onReconnectStart={isDisabled ? undefined : handleReconnectStart}
+          onReconnect={isDisabled ? undefined : handleReconnect}
+          onReconnectEnd={isDisabled ? undefined : handleReconnectEnd}
+          onEdgesDelete={isDisabled ? undefined : onEdgesDelete}
+          proOptions={proOptions}
+          connectionRadius={55}
+          connectionLineType={ConnectionLineType.SmoothStep}
+          connectionLineComponent={EdgeInProgress}
+          isValidConnection={isValidConnection}
+          minZoom={MIN_ZOOM}
+          onlyRenderVisibleElements
+        >
+          <Controls
+            position="top-right"
+            style={{ top: 80 }}
+            showZoom={false}
+            showFitView={false}
+            showInteractive={false}
+          >
+            {!isDisabled && (
```

Comment:
Looks like these are hidden when disabled. Does it make more sense to disable the components themselves?

Reply:
The `showZoom`, `showFitView` and `showInteractive` props of the `reactflow.Controls` component show/hide the built-in (in reactflow) implementation of these features. We have hidden them explicitly, as we are using the same buttons with our own controller logic.

The actual zoom, fit and lock buttons are not rendered when the editor is in disabled mode. (As implemented in the lines following this)